### PR TITLE
Fixed the session timeout prevention function in WebForms.

### PR DESCRIPTION
**Pull request by SandeepNayak077**
- Modified jQuery.get() to $.Ajax() method.
- Added cache property in the Ajax ping request to prevent the session timeout.

### CODE REVIEW

**Comment by daisukenishino** [comment]
@yusukemaegawa 

This is corespondense of the https://github.com/OpenTouryoProject/OpenTouryo/issues/184 issue.

Please review.
I concerned setting of the dataType.
```
dataType: "json"
```

File: root/programs/C#/Samples/WebApp_sample/ProjectX_sample/ProjectX_sample/Framework/Js/common.js [hunk 1]
```diff
@@ -75,9 +76,15 @@ function Fx_Document_OnLoad2() {
...
 //// ---------------------------------------------------------------
 //function HttpPing() {
-//    var httpObj = jQuery.get("/ProjectX_sample/Framework/ping.aspx", null, function () {
-//        //動作確認用コード(ping先の画面のソースをアラート表示)
-////        alert(httpObj.responseText);
+//    $.ajax({
+//        type: 'GET',
+//        url: ResolveServerUrl("~/Aspx/Framework/ping.aspx"),
+//        contentType: "application/json; charset=utf-8",
+//        data: {},
```
Comment (yusukemaegawa) [inline root/programs/C#/Samples/WebApp_sample/ProjectX_sample/ProjectX_sample/Framework/Js/common.js]: @SandeepNayak077 

If no data is sent, I think that this option is unnecessary.

File: root/programs/C#/Samples/WebApp_sample/ProjectX_sample/ProjectX_sample/Framework/Js/common.js [hunk 2]
```diff
@@ -75,9 +76,15 @@ function Fx_Document_OnLoad2() {
 //// 戻り値  －
 //// ---------------------------------------------------------------
 //function HttpPing() {
-//    var httpObj = jQuery.get("/ProjectX_sample/Framework/ping.aspx", null, function () {
-//        //動作確認用コード(ping先の画面のソースをアラート表示)
-////        alert(httpObj.responseText);
+//    $.ajax({
+//        type: 'GET',
+//        url: ResolveServerUrl("~/Aspx/Framework/ping.aspx"),
+//        contentType: "application/json; charset=utf-8",
```
Comment (yusukemaegawa) [inline root/programs/C#/Samples/WebApp_sample/ProjectX_sample/ProjectX_sample/Framework/Js/common.js]: @SandeepNayak077 

If no data is sent, I think that this option is unnecessary.

File: root/programs/C#/Samples/WebApp_sample/ProjectX_sample/ProjectX_sample/Framework/Js/common.js [hunk 3]
```diff
@@ -75,9 +76,15 @@ function Fx_Document_OnLoad2() {
...
+//        data: {},
+//        cache:false,
+//        dataType: "json",
+//        success: function () {},
```
Comment (yusukemaegawa) [inline root/programs/C#/Samples/WebApp_sample/ProjectX_sample/ProjectX_sample/Framework/Js/common.js]: @SandeepNayak077 

If there is not any processing, I think that this option is unnecessary.

File: root/programs/C#/Samples/WebApp_sample/ProjectX_sample/ProjectX_sample/Framework/Js/common.js [hunk 4]
```diff
@@ -75,9 +76,15 @@ function Fx_Document_OnLoad2() {
...
+//        error: function () {}
```
Comment (yusukemaegawa) [inline root/programs/C#/Samples/WebApp_sample/ProjectX_sample/ProjectX_sample/Framework/Js/common.js]: @SandeepNayak077 

If there is not any processing, I think that this option is unnecessary.

File: root/programs/C#/Samples/WebApp_sample/ProjectX_sample/ProjectX_sample/Framework/Js/common.js [hunk 5]
```diff
@@ -75,9 +76,15 @@ function Fx_Document_OnLoad2() {
...
-//    var httpObj = jQuery.get("/ProjectX_sample/Framework/ping.aspx", null, function () {
-//        //動作確認用コード(ping先の画面のソースをアラート表示)
-////        alert(httpObj.responseText);
+//    $.ajax({
+//        type: 'GET',
+//        url: ResolveServerUrl("~/Aspx/Framework/ping.aspx"),
+//        contentType: "application/json; charset=utf-8",
+//        data: {},
+//        cache:false,
+//        dataType: "json",
```
Comment (yusukemaegawa) [inline root/programs/C#/Samples/WebApp_sample/ProjectX_sample/ProjectX_sample/Framework/Js/common.js]: @SandeepNayak077 

I think that this option is unnecessary.

The reasons:
- When sending request to ping.aspx, expected type of data is *html*.
- [This article](http://api.jquery.com/jquery.ajax/) describes as follows:  
"The type of data that you're expecting back from the server. If none is specified, jQuery will try to infer it based on the MIME type of the response (an XML MIME type will yield XML, in 1.4 JSON will yield a JavaScript object, in 1.4 script will execute the script, and anything else will be returned as a string)."

And, an ASPX page streams *text/html* to the browser.
So, even if you do not set this option, jQuery will infer that expected type of data is *html*.

**Comment by SandeepNayak077** [comment]
@daisukenishino 

I have removed unnecessary options.
Please review and merge this pull request.

File: root/programs/C#/Samples/WebApp_sample/ProjectX_sample/ProjectX_sample/Framework/Js/common.js [hunk 6]
```diff
@@ -75,9 +76,10 @@ function Fx_Document_OnLoad2() {
 //// 戻り値  －
 //// ---------------------------------------------------------------
 //function HttpPing() {
-//    var httpObj = jQuery.get("/ProjectX_sample/Framework/ping.aspx", null, function () {
-//        //動作確認用コード(ping先の画面のソースをアラート表示)
-////        alert(httpObj.responseText);
+//    $.ajax({
```
Comment (yusukemaegawa) [inline root/programs/C#/Samples/WebApp_sample/ProjectX_sample/ProjectX_sample/Framework/Js/common.js]: @SandeepNayak077 
@daisukenishino 

When running this program against my computer, the following error occurs:
```
ReferenceError: $ is not defined
```

Where is jQuery referred to?

Comment (SandeepNayak077) [inline root/programs/C#/Samples/WebApp_sample/ProjectX_sample/ProjectX_sample/Framework/Js/common.js]: @daisukenishino 
@yusukemaegawa 

We have to manually add the following jQuery references before running the application
- jquery-ui-*.js 
- jquery.validate.js

This has been explained in the evidence file.
Please confirm.

Comment (daisukenishino) [inline root/programs/C#/Samples/WebApp_sample/ProjectX_sample/ProjectX_sample/Framework/Js/common.js]: @SandeepNayak077 

Get a run-time error of JavaScript in the current implementation.
The message like **"$ is not defined"** is displayed the in this error.
Maybe, **jquery.js (or jquery.min.js)** required.
There is no relationship to **jquery-ui-*.js** and **jquery.validate.js**.

Comment (daisukenishino) [inline root/programs/C#/Samples/WebApp_sample/ProjectX_sample/ProjectX_sample/Framework/Js/common.js]: @SandeepNayak077 

Add the following to the master page.
```HTML
<!--script src="http://code.jquery.com/jquery-1.11.1.min.js"></script-->
```

And, add the link as following.
```
// Sessionタイムアウト防止機能 - Open 棟梁 Wiki
// https://opentouryo.osscons.jp/index.php?Session%E3%82%BF%E3%82%A4%E3%83%A0%E3%82%A2%E3%82%A6%E3%83%88%E9%98%B2%E6%AD%A2%E6%A9%9F%E8%83%BD
// Webサーバへ一定時間ごとにpingを行う
//window.setInterval(HttpPing, 5 * 60 * 1000);
```

Comment (yusukemaegawa) [inline root/programs/C#/Samples/WebApp_sample/ProjectX_sample/ProjectX_sample/Framework/Js/common.js]: @SandeepNayak077 
@daisukenishino 

Please comment as follows:
```HTML
<!--<script src="http://code.jquery.com/jquery-1.11.1.min.js"></script>-->
```

The reason:
Developers can comment or uncomment by using the features of visual studio, such as *comment* / *uncomment* icon in toolbar.

**Comment by SandeepNayak077** [comment]
@daisukenishino 

> https://github.com/SymphonyTeleca/OpenTouryo/pull/140#discussion_r71998731

Yes correct, **jquery-{version}.js** used to resolve the issue.
It was typo error, sorry for the inconvenience.

> https://github.com/SymphonyTeleca/OpenTouryo/pull/140#discussion_r72006355
> https://github.com/SymphonyTeleca/OpenTouryo/pull/140#discussion_r72006944

I have pushed the changes into this pull request.
Please confirm.

File: root/programs/C#/Samples/WebApp_sample/ProjectX_sample/ProjectX_sample/Aspx/Common/sampleScreen.master [hunk 7]
```diff
@@ -8,6 +8,9 @@
 <html xmlns="http://www.w3.org/1999/xhtml" >
 <head runat="server">
     <title>無題のページ</title>
+    <!--Uncomment below line when using jQuery-->
+    <!--<script src="http://code.jquery.com/jquery-1.11.1.min.js"></script>-->
```
Comment (yusukemaegawa) [inline root/programs/C#/Samples/WebApp_sample/ProjectX_sample/ProjectX_sample/Aspx/Common/sampleScreen.master]: @SandeepNayak077 
@daisukenishino 

The version 1.11.1 is old.
Please use the latest version of jquery.

- When supporting Internet Explorer 6 or later
```HTML
<script src="http://code.jquery.com/jquery-1.12.4.min.js"></script>
```

- When supporting Internet Explorer 9 or later  
(*NOT* supporting Internet Explorer 8 or earlier)
```HTML
<script src="http://code.jquery.com/jquery-3.1.0.min.js"></script>
```

The performance of jquery 3.1.0 is better than jquery 1.12.4.

Comment (daisukenishino) [inline root/programs/C#/Samples/WebApp_sample/ProjectX_sample/ProjectX_sample/Aspx/Common/sampleScreen.master]: @SandeepNayak077 

>https://github.com/SymphonyTeleca/OpenTouryo/pull/140#discussion_r72169671

IE is supported version 11 or later.
Therefore, please adapt the following jquery version 3.* series.

```HTML
<script src="http://code.jquery.com/jquery-3.1.0.min.js"></script>
```

**Comment by daisukenishino** [comment]
@SandeepNayak077 
@supragyan 

Please suggest the correspondence order of the following.

- [This pull-request (Fixed the session timeout prevention function in WebForms).](https://github.com/SymphonyTeleca/OpenTouryo/pull/140)
- [Changed the path of ProjectX_sample web application.](https://github.com/SymphonyTeleca/OpenTouryo/pull/143)
- [Converted ProjectX_Sample website to Web application for VB version.](https://github.com/SymphonyTeleca/OpenTouryo/pull/142)

It is desirable that the work is simple.

**Comment by SandeepNayak077** [comment]
@daisukenishino 

> It is desirable that the work is simple.

As discussed with team, we concluded that  
[1. Changed the path of ProjectX_sample web application.](https://github.com/SymphonyTeleca/OpenTouryo/pull/140)
This has large number of file change, so please merge this first.

[2. Fixed the session timeout prevention function in WebForms. #140](https://github.com/SymphonyTeleca/OpenTouryo/pull/140)
Since this correspondence has less change, so, we will recreate the feature branch without affecting any.  

[3. Converted ProjectX_Sample website to Web application for VB version.](https://github.com/SymphonyTeleca/OpenTouryo/pull/142)
We think that there will not be any conflict in this pull request because the files are different.
However, please do not merge this pull request, as the application folder path needs to be changed.
We will send new pull request once completed. 

Please confirm.

File: root/programs/C#/Samples/WebApp_sample/ProjectX_sample/ProjectX_sample/Aspx/Common/sampleScreen.master (hunk 7 above)
Comment (SandeepNayak077) [inline root/programs/C#/Samples/WebApp_sample/ProjectX_sample/ProjectX_sample/Aspx/Common/sampleScreen.master]: @daisukenishino 

I have updated this pull request with the above changes.
Please let us know.

Comment (daisukenishino) [inline root/programs/C#/Samples/WebApp_sample/ProjectX_sample/ProjectX_sample/Aspx/Common/sampleScreen.master]: @SandeepNayak077 

I see.
I confirmed.
https://github.com/SymphonyTeleca/OpenTouryo/pull/140/files

**Comment by daisukenishino** [comment]
@SandeepNayak077 

I see. I will do so.

**Comment by daisukenishino** [comment]
@SandeepNayak077 

>[1. Changed the path of ProjectX_sample web application.](https://github.com/SymphonyTeleca/OpenTouryo/pull/140)
This has large number of file change, so please merge this first.

This has been completed.

>[2. Fixed the session timeout prevention function in WebForms. #140](https://github.com/SymphonyTeleca/OpenTouryo/pull/140)
Since this correspondence has less change, we will recreate the feature branch without affecting any.  

Next, plese do this corespondence.

**Pull request merged**